clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslClientCallbackHandler.java

```diff
@@ -70,6 +71,8 @@ public void handle(Callback[] callbacks) throws IOException, UnsupportedCallback
         for (Callback callback : callbacks) {
             if (callback instanceof OAuthBearerTokenCallback)
                 handleCallback((OAuthBearerTokenCallback) callback);
+            else if (callback instanceof SaslExtensionsCallback)
```

Comment:
Please update Javadoc to state that this class also recognizes {@link SaslExtensionsCallback} and retrieves any SASL extensions that were created when the {@code OAuthBearerLoginModule} logged in by looking for an instance of {@code Map} in the {@code Subject}'s public credentials. (or an instance of SaslExtensions rather than a Map if we decide to make SaslExtensions part of the public API)